proposed/http-client/http-client-meta.md

```diff
@@ -33,6 +33,11 @@ client will allow libraries to be decoupled from an implementation such as Guzzl
 specify the default behaviours.
 * The purpose is not to be opinionated about the use of middlewares (PSR-15).
 
+#### Asynchronous HTTP client
+
+The reason asynchronous support is a non-goal is because a PSR for HTTP clients 
+should not define Promises. That should be a separate PSR. Waiting for a such PSR 
```

Comment:
Nowhere in the PSR-7 specification does it say that `getContents` has to use `__toString`. (In fact `getContents` and `__toString` has different behaviours. ) Nor does it say that it has to be blocking. 

Im saying that you **can** implement PSR-7 with non-blocking. It is not a limitation in the interface but in the implementations.